Dockerfile

```diff
@@ -9,12 +9,13 @@ RUN apk --update upgrade && \
 ADD . /go/src/code-annotation
 WORKDIR /go/src/code-annotation
 
-RUN make build && \
+RUN make build-frontend && \
```

Comment:
You should copy the file from the `bin` folder like:
https://github.com/src-d/ci/blob/v1/examples/basic/Dockerfile#L2

And then you can remove almost the full Dockefile

Reply:
this way allows to build docker file on any machine without installing any dependencies like go or nodejs. It was requirement from @vmarkovtsev to be able to build docker file without correct version of nodejs and yarn

Reply:
Yes, I can confirm that: I even have no idea what is `yarn` and why it is better than `npm`

Reply:
Err but if you are contributing you need to run the tests and other stuff. So you need yarn installed anyway. 

Anyways you need to call `make packages` for having the proper version variables.

Reply:
If contributing - agree 100%. If building the container to run the thing and forget about it - nopes.

Reply:
Also, you have the contributing guide, were is explained. 

At bblfsh, all the tests are executed under a docker images, but the makefile executes all the task inside of the dockerfile and not the opposite like here. Is done in this way, because of the nature of bblfsh.

But here, if you are going to contribute to this project you should have to run the test, and you can't if yarn isn't installed.

Reply:
don't mess up CI and release with local building of container.

Reply:
example use case: I'm a devops. I want to run deploy this tool on my cluster. There is a change in master I need. But no release yet. So I can just `git clone && docker build -t my/tool . && docker push my/tool && kubectl <deploy>`

Reply:
Errr, no.  You shouldn't, never do this, you should our standard image, tagged and tests.  This Dockerfile is for the CI to build docker images.

Reply:
But I am not understanding this, discussion. We agreed and merged a CI, to be used, and not to skip it, and the goal of CI is exactly this one, having homogeneous repositories.

Reply:
okay, will remove build step from Dockerfile. I'm just used to be able to build images without external dependencies.

P.S. example of our app (merged by you):
https://github.com/src-d/github-reminder/blob/master/Dockerfile